stl/inc/memory

```diff
@@ -584,12 +584,18 @@ namespace ranges {
     template <_No_throw_input_iterator _It, _No_throw_sentinel_for<_It> _Se>
         requires destructible<iter_value_t<_It>>
     _NODISCARD constexpr _It _Destroy_unchecked(_It _First, _Se _Last) noexcept {
-        if constexpr (is_trivially_destructible_v<iter_value_t<_It>>) {
-            _RANGES advance(_First, _STD move(_Last));
-        } else {
+        if consteval {

```

Comment:
I'm not sure.

On one hand I'd prefer to avoid code repetition like this:
```c++
if !consteval {
    if constexpr (...) {
       optimized;
       early-return;
   }
}
unoptimized;
```

On the other, the way it is written is cleaner control flow.

Reply:
I took the pattern proposed in https://github.com/microsoft/STL/issues/5225#issuecomment-2578815415, except that it isn't worthy to me to introduce new helper functions.

Reply:
Code duplication bothers me but dead code bothers me slightly more, so I'm generally fine with this as long as the code is short enough that a helper function wouldn't be worth it.